### PR TITLE
Add docs for Microsoft.Extensions.DependencyModel.Library and subclasses

### PR DESCRIPTION
**Pull request by MSDN-WhiteKnight**
Issue: #3418 

Add API docs for Microsoft.Extensions.DependencyModel.*Library classes and example for CompilationLibrary. Based on [.deps.json specification](https://github.com/dotnet/sdk/blob/main/documentation/specs/runtime-configuration-file.md) and source code.

### CODE REVIEW

**Comment by BillWagner** [comment]
ping @jeffhandley 

There isn't a label in this repo for the "area-DependencyModel" to help fabric bot. How should we get the correct label and reviewers assigned?

**Comment by Jozkee** [comment]
@eiriktsarpalis @joperezr, do you know how to add the missing label @BillWagner mentions?

**Comment by eiriktsarpalis** [comment]
It should take a couple of steps:

1. Creating the area label manually for the repo.
2. Add an entry in the [corresponding rule](https://github.com/dotnet/dotnet-api-docs/blob/0d185923a4b6e043f1b65428f130a7519a46aabb/.github/fabricbot.json#L53-L522) in the repo's `fabricbot.json` file. If memory serves this rule is being updated manually, but if not there might be a relevant script in the [fabricbot-config](https://github.com/dotnet/fabricbot-config) repo. @jeffhandley or @ericstj might know for sure.

**Comment by ericstj** [comment]
It only requires the first step as it looks like the fabricbot configuration already accounts for the labels.  Anyone with write permission in the repo can create a label.  I just did so and applied it here.

**Review by ViktorHofer — APPROVED** [review]

File: snippets/csharp/Microsoft.Extensions.DependencyModel/CompilationLibrary/Overview/CompilationLibrarySnippets.csproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (ViktorHofer) [inline snippets/csharp/Microsoft.Extensions.DependencyModel/CompilationLibrary/Overview/CompilationLibrarySnippets.csproj:5]:
```suggestion
    <TargetFramework>net8.0</TargetFramework>
```

File: snippets/csharp/Microsoft.Extensions.DependencyModel/CompilationLibrary/Overview/CompilationLibrarySnippets.csproj [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    <PreserveCompilationContext>true</PreserveCompilationContext>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="6.0.0" />
```
Comment (ViktorHofer) [inline snippets/csharp/Microsoft.Extensions.DependencyModel/CompilationLibrary/Overview/CompilationLibrarySnippets.csproj:10]:
```suggestion
    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="8.0.0" />
```

**Comment by ViktorHofer** [comment]
@gewarren and @MSDN-WhiteKnight sorry for the long delay, I didn't get pinged on this until recently. I just reviewed and the doc additions look great. Thanks a lot @MSDN-WhiteKnight.

**Pull request merged**